Add possibility to disable label reconciliation (#2486)
gravitational/gravity: docs/9.x/cluster.md

      effect: NoExecute
```

## Custom Labels

Gravity supports provisioning the labels for the kubernetes nodes. For assignment use the `labels` property in the node profiles:

```yaml
nodeProfiles:
  - name: node
    description: "Gravity Node"
    labels:
      label1: value1
      label2: value2
```

Gravity has three modes of label reconciliation, which the user can change at any time:

>  - `EnsureExists` (default) - Compares the labels in the node with what is written in the profile, the value of the labels is not compared.

Reviewer: ```suggestion
 - `EnsureExists` (default) - Ensures that the labels from the node profile exist recreating the missing ones. The label values from the node profile are not enforced.
```

   - Users can edit the labels as they want.
   - Users cannot delete the label specified in the node profile.

>  - `Enabled` or `true` - Compares the labels in the node with what is written in the profile, the values of the labels are also compared.

Reviewer: ```suggestion
 - `Enabled` or `true` - Ensures that the labels from the node profile exist recreating the missing ones. Also ensures that the label values match those from the node profile.
```

   - Users cannot change the labels specified in the node profile.
   - Users cannot delete the label specified in the node profile.
 - `Disabled` or `false` - Reconciliation of labels is disabled.
   - Users can edit the labels as they want.
   - Users can delete labels.

> Reconciliation mode is assigned to the node and controlled by the label `label-reconciler.gravitational.io/mode`.

Reviewer: ```suggestion
Reconciliation mode on a node is controlled via the label named `label-reconciler.gravitational.io/mode`.
```
